packages/core/src/App/Containers/Modals/app-modals.jsx

```diff
@@ -218,7 +222,16 @@ const AppModals = observer(() => {
         ComponentToLoad = <WalletsUpgradeLogoutModal />;
     }
 
-    if (should_show_effortless_login_modal && !is_tour_open && !is_from_derivgo && !is_onboarding) {
+    if (
+        should_show_effortless_login_modal &&
+        !is_tour_open &&
+        !is_from_derivgo &&
+        !is_onboarding &&
+        !should_show_appropriateness_warning_modal &&
+        !should_show_risk_warning_modal &&
+        !is_from_signup_account &&
+        !is_real_acc_signup_on
```

Comment:
@yauheni-deriv i was also thinking instead of adding all these conditions just to hide the modal during mf account onboarding, what if set a observer true when user is in onboarding flow and set to false when user leaves the onboarding flow. So we just hide the **_EffortlessLoginModal_** when that observer is true.

Reply:
after our discussion: we can think on refactoring later as it is prod bug